### PR TITLE
Update mag phase

### PR DESCRIPTION
**Pull request by dmdmello**
Finished updating the Branch containing all the changes regarding filter verification used for the sbrt17 conference. I believe there are no conflicts whatsoever.

### CODE REVIEW

File: bmc/core/compatibility.h [hunk 1]
```diff
@@ -36,7 +36,14 @@ void __DSVERIFIER_assert(_Bool expression){
 	assert(expression);
 }
 
-void __DSVERIFIER_assert_msg(_Bool expression, char * msg){
-	printf("%s", msg);
+void __DSVERIFIER_assert_msg(_Bool expression, char  msg[]){
+	/*
```
Comment (lucasccordeiro) [inline bmc/core/compatibility.h]: Can you please remove those comments from the code?

File: bmc/core/filter_functions.h [hunk 2]
```diff
@@ -57,9 +59,14 @@ double sinTyl(double x, int precision){
                                 aux = aux*xsquared;
                                 sine -=aux/39916800;
                                 if (precision >= 7)
-                                  printf("Warning: Function sinTyl "
-                                  "from bmc/core/filter_functions.h: Precision "
-                                  "representation exceeded. Assuming maximum precision of 6\n");
+                                {
```
Comment (lucasccordeiro) [inline bmc/core/filter_functions.h:62]: Can you please fix the code indentation?

File: bmc/core/fixed-point.h [hunk 3]
```diff
@@ -219,12 +219,12 @@ fxp_t fxp_double_to_fxp(double value) {
 	fxp_t tmp;
 	double ftemp = value * scale_factor[impl.frac_bits];
 	if (rounding_mode == ROUNDING){
-		if(value >= 0) {
-			tmp = (fxp_t)(ftemp + 0.5);
-		}
-		else {
-			tmp = (fxp_t)(ftemp - 0.5);
-		}
+		//if(value >= 0) {
```
Comment (lucasccordeiro) [inline bmc/core/fixed-point.h]: Can you please remove the code that is commented?

File: bmc/core/util.h [hunk 4]
```diff
@@ -17,6 +17,8 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdbool.h>
+ #include <math.h>
```
Comment (lucasccordeiro) [inline bmc/core/util.h]: Remove whitespace

File: bmc/core/util.h [hunk 5]
```diff
@@ -591,3 +593,28 @@ void fxp_transpose(fxp_t a[LIMIT][LIMIT], fxp_t b[LIMIT][LIMIT],int n, int m)
...
+
+
+bool doubleComparisson(double a, double b, double epslon)
+{
+    return fabs(a - b) < epslon;
+}
+
+
+double fabs_(double a){
+
```
Comment (lucasccordeiro) [inline bmc/core/util.h]: Remove empty line

File: bmc/core/util.h [hunk 6]
```diff
@@ -591,3 +593,28 @@ void fxp_transpose(fxp_t a[LIMIT][LIMIT], fxp_t b[LIMIT][LIMIT],int n, int m)
...
+	if (a < 0) return a * (-1.0);
+	
+	else return a; 
+
+}
+/*
+double fabs2(double num){
```
Comment (lucasccordeiro) [inline bmc/core/util.h]: remove code that is commented

File: bmc/dsverifier.h [hunk 7]
```diff
@@ -43,7 +43,7 @@
 #include "engine/verify_controllability.h"
 #include "engine/verify_observability.h"
 #include "engine/verify_magnitude.h"
-
```
Comment (lucasccordeiro) [inline bmc/dsverifier.h:48]: Remove empty line

File: bmc/engine/verify_magnitude.h [hunk 8]
```diff
@@ -52,67 +59,223 @@ void resp_mag(double* num, int lnum, double* den, int lden, double* res, int N)
 			out_denIm[i] = sinTyl(w, SINE_precision) * old_out_Re + cosTyl(w, SINE_precision) * out_denIm[i];
 		}
 
-		res[i] = sqrt3(out_numRe[i] * out_numRe[i] + out_numIm[i] * out_numIm[i]); 
-	    zero_test = sqrt3(out_denRe[i] * out_denRe[i] + out_denIm[i] * out_denIm[i]);
-	    __DSVERIFIER_assume(zero_test != 0);
-		res[i] = res[i] / zero_test;
+		res[i] = sqrt(out_numRe[i] * out_numRe[i] + out_numIm[i] * out_numIm[i]);
```
Comment (lucasccordeiro) [inline bmc/engine/verify_magnitude.h:61]: fix indentation

File: bmc/engine/verify_magnitude.h [hunk 9]
```diff
@@ -52,67 +59,223 @@ void resp_mag(double* num, int lnum, double* den, int lden, double* res, int N)
...
+	    //__DSVERIFIER_assume(zero_test != 0);
+		res[i] = res[i] /zero_test;
+ 			
+
+
+	}
+}
+
+/*
+void resp_mag(double* num, int lnum, double* den, int lden, double* res, int N) {
```
Comment (lucasccordeiro) [inline bmc/engine/verify_magnitude.h]: remove code that is commented

File: bmc/engine/verify_magnitudeESBMC.h [hunk 10]
```diff
@@ -0,0 +1,398 @@
...
+ /* printf("B[0]=%.20f \n", ds.b[0]);
+  printf("B[0]=%.20f \n", ds.b[1]);
+  printf("B[0]=%.20f \n", ds.b[2]);
+
+  printf("B[0]=%.20f \n", ds.a[0]);
+  printf("B[0]=%.20f \n", ds.a[1]);
+  printf("B[0]=%.20f \n", ds.a[2]);
+*/
+
+ printf("----------------------------Resposta em PONTO FLUTUANTE----------------------------\n");
```
Comment (lucasccordeiro) [inline bmc/engine/verify_magnitudeESBMC.h]: translate text to English!

File: bmc/engine/verify_magnitudeESBMC.h [hunk 11]
```diff
@@ -0,0 +1,398 @@
...
+   // ds.b[i] = fxp_to_float(fxp_float_to_fxp(ds.b[i]));
+    printf("B[%d]=%.20f \n", i, ds.b[i]);
+  }
+  for (i=0; i<ds.a_size; i++) {
+   // ds.a[i] = fxp_to_float(fxp_float_to_fxp(ds.a[i]));
+    printf("A[%d]=%.20f \n", i, ds.a[i]);
+  }
+
+*/
+  printf("----------------------------Resposta em PONTO FIXO----------------------------\n");
```
Comment (lucasccordeiro) [inline bmc/engine/verify_magnitudeESBMC.h]: You should write in English!

**Review by lucasccordeiro — COMMENTED** [review]
- fix indentation
- remove code that is commented
- write your comments, etc in English

File: bmc/core/fixed-point.h (hunk 3 above)
Comment (ssvlab) [reply bmc/core/fixed-point.h]: @dmdmello  Why did you remove this code about the rounding mode?

File: bmc/dsverifier.h [hunk 12]
```diff
@@ -358,117 +362,6 @@ void call_verification_task(void * verification_task)
...
-				{
-					base_case_executed = 1;
-				}
-
-				ds.a[i] = nondet_double();
-				__DSVERIFIER_assume((ds.a[i] >= min) && (ds.a[i] <= max));
-			}
-		}
-	}
-
```
Comment (ssvlab) [inline bmc/dsverifier.h]: @dmdmello Why did you remove this code about error mode?

File: bmc/engine/verify_magnitudeESBMC.h (hunk 10 above)
Comment (ssvlab) [reply bmc/engine/verify_magnitudeESBMC.h]: and remove the commented code. :)

File: bmc/engine/verify_phase.h [hunk 13]
```diff
@@ -0,0 +1,131 @@
+/**
+ * DSVerifier - Digital Systems Verifier
```
Comment (ssvlab) [inline bmc/engine/verify_phase.h:2]: Please, remove the commented code. @dmdmello

File: regression/filters/magnitude_det_FIR/filter_03/input.c [hunk 14]
```diff
@@ -0,0 +1,29 @@
+//fhp10Equi
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_03/input.c]: @dmdmello Please, remove the commented code.

File: regression/filters/magnitude_det_FIR/filter_04/input.c [hunk 15]
```diff
@@ -0,0 +1,28 @@
+//fhp30
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_04/input.c]: @dmdmello Please, remove the commented code.

File: regression/filters/magnitude_det_FIR/filter_05/input.c [hunk 16]
```diff
@@ -0,0 +1,25 @@
+//fhp30Equi
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_05/input.c]: @dmdmello Please, remove the commented code.

File: regression/filters/magnitude_det_FIR/filter_06/input.c [hunk 17]
```diff
@@ -0,0 +1,28 @@
+//fhp30Hann
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_06/input.c]: @dmdmello Please, remove the commented code.

File: regression/filters/magnitude_det_FIR/filter_07/input.c [hunk 18]
```diff
@@ -0,0 +1,28 @@
+//flp10
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_07/input.c]: @dmdmello Please, remove the commented code.

File: regression/filters/magnitude_det_FIR/filter_09/input.c [hunk 19]
```diff
@@ -0,0 +1,29 @@
+//flp10EST
+
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_09/input.c]: @dmdmello Please, remove the commented code.

File: regression/filters/magnitude_det_FIR/filter_11/input.c [hunk 20]
```diff
@@ -0,0 +1,29 @@
+//flp10ESTHann
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_11/input.c]: @dmdmello Please, remove the commented code.

File: regression/filters/magnitude_det_FIR/filter_12/input.c [hunk 21]
```diff
@@ -0,0 +1,29 @@
+//flp10Hann
+
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_12/input.c]: @dmdmello Please, remove the commented code.

File: regression/filters/magnitude_det_FIR/filter_13/input.c [hunk 22]
```diff
@@ -0,0 +1,29 @@
+//flp30
+
+#include <dsverifier.h>
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_13/input.c]: @dmdmello Please, remove the commented code.

File: regression/filters/magnitude_det_FIR/filter_15/input.c [hunk 23]
```diff
@@ -0,0 +1,26 @@
+//flp30EST
+
+#include <dsverifier.h>
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_15/input.c]: @dmdmello Please, remove the commented code.

File: regression/filters/magnitude_det_FIR/filter_15/input.c [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+
+digital_system ds = {
+    .b = {    0.030864419451274,   0.031139391134262,   0.031396686913013,   0.031636034832819,   0.031857181621137,   0.032059893020472, 0.032243954096483,   0.032409169520841,   0.032555363828437,   0.032682381648546,   0.032790087909614,   0.032878368017350, 0.032947128005880,   0.032996294661721,   0.033025815620389,   0.033035659435524,   0.033025815620389,   0.032996294661721, 0.032947128005880,   0.032878368017350,   0.032790087909614,   0.032682381648546,   0.032555363828437,   0.032409169520841, 0.032243954096483,  0.032059893020472,   0.031857181621137,   0.031636034832819,   0.031396686913013,   0.031139391134262, 0.030864419451274 },
+    .b_size = 31,
+    .a = {1.0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+    .a_size = 31
+};
+
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_15/input.c]: @dmdmello Please, remove the commented code.

File: regression/filters/magnitude_det_FIR/filter_17/input.c [hunk 25]
```diff
@@ -0,0 +1,26 @@
+//flp30ESTHann
+
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_17/input.c]: @dmdmello Please, remove the commented code.

File: regression/filters/magnitude_det_FIR/filter_18/input.c [hunk 26]
```diff
@@ -0,0 +1,29 @@
...
+digital_system ds = {
+    .b = {   0,  -0.000236155068610,  -0.000621846227125,   0.001488168191543,   0.004550824530708,  -0.000000000000000,-0.011615851114965,  -0.010466472551074,   0.014754259300053,   0.033008076298194,  -0.000000000000000,  -0.063133011996516, -0.056384525736815,   0.089463797278168,   0.299284946863305,   0.399815580466267,   0.299284946863305,   0.089463797278168, -0.056384525736815,  -0.063133011996516,  -0.000000000000000,   0.033008076298194,   0.014754259300053,  -0.010466472551074, -0.011615851114965,  -0.000000000000000,  0.004550824530708,   0.001488168191543,  -0.000621846227125,  -0.000236155068610, 0 },
+    .b_size = 31,
+    .a = {1.0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+    .a_size = 31
+};
+
+filter_parameters filter = {
+    .Ac = 0.501187233627272,
+ //   .Ar = 0.501187233627272,
```
Comment (ssvlab) [inline regression/filters/magnitude_det_FIR/filter_18/input.c]: @dmdmello Please, remove the commented code.

File: src/dsverifier.cpp [hunk 27]
```diff
@@ -848,7 +891,7 @@ std::string prepare_bmc_command_line()
 	  command_line = "gcc -E " + desired_filename + " -DK_INDUCTION_MODE=K_INDUCTION -DBMC=ESBMC -I " + bmc_path;
 	} else {
           command_line = model_checker_path + "/esbmc " + desired_filename +
-	  " --no-bounds-check --no-pointer-check --no-div-by-zero-check -DBMC=ESBMC -I " +
```
Comment (ssvlab) [inline src/dsverifier.cpp:852]: @dmdmello we need to use the flag "--no-div-by-zero-check" in order to avoid the divisions by zero during the verification.

File: src/dsverifier.cpp [hunk 28]
```diff
@@ -1636,23 +1682,35 @@ void generates_mag_response(double* num, int lnum, double* den, int lden, double
...
 		}
 
 		res[i] = sqrt(out_numRe[i] * out_numRe[i] + out_numIm[i] * out_numIm[i]); 
 	    zero_test = sqrt(out_denRe[i] * out_denRe[i] + out_denIm[i] * out_denIm[i]);
-	 	//if (zero_test != 0)
-		res[i] = res[i] / zero_test;
+		res[i] = res[i] /zero_test;
+	}
+
+	//If necessary, uncomment the following "for" statement for debug porpouses
```
Comment (ssvlab) [inline src/dsverifier.cpp]: @dmdmello Please, remove the commented code.

File: src/dsverifier.cpp [hunk 29]
```diff
@@ -1724,30 +1822,114 @@ void check_filter_magnitude_det()
...
+				if(!(_res[i] >= filter.Ap))
 				{
 					printf("|----------------Passband Failure-------------|");
 					response_is_valid = false;
 					break;
 				} 
 			}
 		}
+	} else if (filter.type == PASSBAND) { //passband
+
```
Comment (ssvlab) [inline src/dsverifier.cpp]: @dmdmello Please, remove the commented code.

File: src/dsverifier.cpp (hunk 27 above)
Comment (dmdmello) [reply src/dsverifier.cpp:852]: Sorry, I was trying to detect a division by zero in my code and I deleted it. I'm going to write it back.

File: bmc/engine/verify_magnitudeESBMC.h (hunk 11 above)
Comment (dmdmello) [reply bmc/engine/verify_magnitudeESBMC.h]: Sorry, It was an accident.

File: bmc/engine/verify_magnitudeESBMC.h (hunk 10 above)
Comment (dmdmello) [reply bmc/engine/verify_magnitudeESBMC.h]: Sorry, It was an accident.

File: regression/filters/magnitude_det_FIR/filter_03/input.c (hunk 14 above)
Comment (dmdmello) [reply regression/filters/magnitude_det_FIR/filter_03/input.c]: Sorry, I can't delete these comments at the beginning of each filter in the regressions folder. They are there to identify which filter was used. I'd better delete the entire filters regression folder, if that's the case.

File: bmc/core/fixed-point.h (hunk 3 above)
Comment (dmdmello) [reply bmc/core/fixed-point.h]: It was an accident. I was trying to see whether my program was using it or not.

File: bmc/core/definitions.h [hunk 30]
```diff
@@ -229,7 +231,9 @@ typedef struct {
 } hardware;
 
 typedef struct{
-  float Ap, Ar, Ac;
-  float wp, wc, wr; 
+  double Ap, Ar, Ac;
```
Comment (feliperodri) [inline bmc/core/definitions.h]: Could you please add comments to identify this struct?

File: bmc/engine/verify_magnitude.h (hunk 9 above)
Comment (feliperodri) [reply bmc/engine/verify_magnitude.h]: @dmdmello could you also please remove unnecessary empty lines?

File: bmc/engine/verify_magnitude.h [hunk 31]
```diff
@@ -52,67 +59,223 @@ void resp_mag(double* num, int lnum, double* den, int lden, double* res, int N)
...
+	//Ac_margin = diff;
 
 /*
  * Magnitude verifier 
  */
  int verify_magnitude(void) {
 
-	int freq_response_samples = 100;
+ 	rounding_mode = ROUNDING;
+
```
Comment (feliperodri) [inline bmc/engine/verify_magnitude.h:73]: remove unnecessary empty lines

File: bmc/engine/verify_magnitude.h [hunk 32]
```diff
@@ -52,67 +59,223 @@ void resp_mag(double* num, int lnum, double* den, int lden, double* res, int N)
...
+
+	int freq_response_samples = 100;	
 	double w;
 	double w_incr = 1.0 / freq_response_samples;
-  	double res[freq_response_samples+1];
+  	//double res[freq_response_samples+1];
```
Comment (feliperodri) [inline bmc/engine/verify_magnitude.h]: remove commented code

File: bmc/engine/verify_magnitude.h [hunk 33]
```diff
@@ -52,67 +59,223 @@ void resp_mag(double* num, int lnum, double* den, int lden, double* res, int N)
...
+  	double _res[freq_response_samples+1];
   	int i,j;
+  	char msg[50]; 
 
-  	/* quantize "a" array using fxp */
+	/* generates magnitude response, placing the result in the "res" array*/
+	//resp_mag(ds.b, ds.b_size, ds.a, ds.a_size, res, freq_response_samples);
```
Comment (feliperodri) [inline bmc/engine/verify_magnitude.h]: remove commented code

File: bmc/engine/verify_magnitude.h [hunk 34]
```diff
@@ -52,67 +59,223 @@ void resp_mag(double* num, int lnum, double* den, int lden, double* res, int N)
...
-	fxp_double_to_fxp_array(ds.b, b_fxp, ds.b_size);
-	double _b[ds.b_size];
-	fxp_to_double_array(_b, b_fxp, ds.b_size);
 
- /* generates magnitude response of the quantized TF, placing the result in the "res" array*/
+	/* generates magnitude response of the quantized TF, placing the result in the "_res" array*/
+	resp_mag(ds.b, ds.b_size, ds.a, ds.a_size, _res, freq_response_samples);
 
-  resp_mag(ds.b, ds.b_size, ds.a, ds.a_size, res, freq_response_samples);
+ 	//double res_Ac;
```
Comment (feliperodri) [inline bmc/engine/verify_magnitude.h]: remove commented code

File: bmc/engine/verify_magnitude.h [hunk 35]
```diff
@@ -52,67 +59,223 @@ void resp_mag(double* num, int lnum, double* den, int lden, double* res, int N)
...
+	//if (Ac_margin < 0) {
+	//	double aux = Ac_margin * (-1);
+	//	Ac_margin = aux;
+	//}
+
+
+	//double Ac_margin = 0;
+
+	if ((filter.wp == 0) && (filter.wr == 0)){
+
```
Comment (feliperodri) [inline bmc/engine/verify_magnitude.h]: remove unnecessary empty l

File: bmc/engine/verify_magnitude.h [hunk 36]
```diff
@@ -52,67 +59,223 @@ void resp_mag(double* num, int lnum, double* den, int lden, double* res, int N)
...
+			filter.wr = filter.wc + w_incr;
+		}
 
+		if (filter.type == HIGHPASS){
+			filter.wp = filter.wc + w_incr;
+			filter.wr = filter.wc - w_incr;
+		}
+	}
+
+/*
```
Comment (feliperodri) [inline bmc/engine/verify_magnitude.h]: remove commented code

File: bmc/engine/verify_magnitudeESBMC.h [hunk 37]
```diff
@@ -0,0 +1,398 @@
...
+#define SINE_precision 6
+#define LOWPASS 1
+#define HIGHPASS 2
+
+/*
+ *  Generates magnitude response from transfer function
+ */
+void resp_mag(double* num, int lnum, double* den, int lden, double* res, int N) {
+
+	//__ESBMC_assume(N<110);
```
Comment (feliperodri) [inline bmc/engine/verify_magnitudeESBMC.h]: remove commented code

File: bmc/engine/verify_magnitudeESBMC.h [hunk 38]
```diff
@@ -0,0 +1,398 @@
...
+		out_denIm[i] = 0;
+
+		for (m = 1; m < lden; ++m) {
+			old_out_Re = out_denRe[i];
+			out_denRe[i] = cosTyl(w, SINE_precision) * out_denRe[i] - sinTyl(w, SINE_precision) * out_denIm[i] + den[m];
+			out_denIm[i] = sinTyl(w, SINE_precision) * old_out_Re + cosTyl(w, SINE_precision) * out_denIm[i];
+		}
+
+		res[i] = sqrt3(out_numRe[i] * out_numRe[i] + out_numIm[i] * out_numIm[i]); 
+	    zero_test = sqrt3(out_denRe[i] * out_denRe[i] + out_denIm[i] * out_denIm[i]);
```
Comment (feliperodri) [inline bmc/engine/verify_magnitudeESBMC.h]: fix indentation

File: bmc/engine/verify_magnitudeESBMC.h [hunk 39]
```diff
@@ -0,0 +1,398 @@
...
+	    __DSVERIFIER_assume(zero_test != 0);
+		res[i] = res[i] / zero_test;
+
+	}
+}
+
```
Comment (feliperodri) [inline bmc/engine/verify_magnitudeESBMC.h]: remove unnecessary empty lines

File: bmc/engine/verify_magnitudeESBMC.h [hunk 40]
```diff
@@ -0,0 +1,398 @@
...
+ int verify_magnitude(void) {
+
+	int N = 100;
+	double w;
+	double w_incr = 1.0 / N;
+  	double res[N+1];
+  	int i,j;
+
+  printf("----------------------------Coeficientes em PONTO FLUTUANTE----------------------------\n");
+ /* printf("B[0]=%.20f \n", ds.b[0]);
```
Comment (feliperodri) [inline bmc/engine/verify_magnitudeESBMC.h]: remove commented code

File: bmc/engine/verify_magnitudeESBMC.h [hunk 41]
```diff
@@ -0,0 +1,398 @@
...
+
+ int verify_magnitude(void) {
+
+	int N = 100;
+	double w;
+	double w_incr = 1.0 / N;
+  	double res[N+1];
+  	int i,j;
+
+  printf("----------------------------Coeficientes em PONTO FLUTUANTE----------------------------\n");
```
Comment (feliperodri) [inline bmc/engine/verify_magnitudeESBMC.h]: Everything must be in English

File: bmc/engine/verify_phase.h [hunk 42]
```diff
@@ -0,0 +1,131 @@
...
+    
+      assert(fabs(res[i]-fi[i]) <= dif);
+  }
+
+}
+/*
+ * Magnitude verifier 
+ */
+
+/*
```
Comment (feliperodri) [inline bmc/engine/verify_phase.h]: duplicated

File: bmc/core/util.h (hunk 5 above)
Comment (lennonchaves) [reply bmc/core/util.h]: @dmdmello, could you please implement @lucasccordeiro comments? Many thanks.

File: bmc/engine/verify_magnitude.h (hunk 8 above)
Comment (lennonchaves) [reply bmc/engine/verify_magnitude.h:61]: @dmdmello, could you please implement @lucasccordeiro comments? Many thanks.

File: bmc/engine/verify_magnitude.h (hunk 32 above)
Comment (lennonchaves) [reply bmc/engine/verify_magnitude.h]: @dmdmello, could you please implement @feliperodri  comments? Many thanks.

File: src/dsverifier.cpp (hunk 27 above)
Comment (lennonchaves) [reply src/dsverifier.cpp:852]: Many thanks! :+1:

File: src/dsverifier.cpp [hunk 43]
```diff
@@ -1636,23 +1682,35 @@ void generates_mag_response(double* num, int lnum, double* den, int lden, double
...
+			out_denIm[i] = sinTyl(w, SINE_precision) * old_out_Re + cosTyl(w, SINE_precision) * out_denIm[i];
 		}
 
 		res[i] = sqrt(out_numRe[i] * out_numRe[i] + out_numIm[i] * out_numIm[i]); 
 	    zero_test = sqrt(out_denRe[i] * out_denRe[i] + out_denIm[i] * out_denIm[i]);
-	 	//if (zero_test != 0)
-		res[i] = res[i] / zero_test;
+		res[i] = res[i] /zero_test;
 	}
+
```
Comment (lennonchaves) [inline src/dsverifier.cpp]: @dmdmello, could you please remove commented code? Many thanks

Comment (dmdmello) [reply src/dsverifier.cpp]: I've removed the commented code, even thought I still think it's useful to just leave it there. As I stated in the comment, those commented lines were meant to be "decommented" in case of debugging purposes.

File: bmc/engine/verify_magnitude.h (hunk 8 above)
Comment (dmdmello) [reply bmc/engine/verify_magnitude.h:61]: FIxed

**Review by ssvlab — APPROVED** [review]
